src/java.base/share/classes/java/lang/VirtualThread.java

```diff
@@ -1444,40 +1448,54 @@ private static ForkJoinPool createDefaultScheduler() {
     /**
      * Schedule a runnable task to run after a delay.
      */
-    private static Future<?> schedule(Runnable command, long delay, TimeUnit unit) {
-        long tid = Thread.currentThread().threadId();
-        int index = (int) tid & (DELAYED_TASK_SCHEDULERS.length - 1);
-        return DELAYED_TASK_SCHEDULERS[index].schedule(command, delay, unit);
+    private Future<?> schedule(Runnable command, long delay, TimeUnit unit) {
+        if (scheduler instanceof ForkJoinPool pool) {
+            return pool.schedule(command, delay, unit);
+        } else {
+            return DelayedTaskSchedulers.schedule(command, delay, unit);
+        }
     }
 
     /**
-     * Creates the ScheduledThreadPoolExecutors used to execute delayed tasks.
+     * Supports scheduling a runnable task to run after a delay. It uses a number
+     * of ScheduledThreadPoolExecutor instances to reduce contention on the delayed
+     * work queue used. This class is used when using a custom scheduler.
```

Comment:
@AlanBateman It might make sense to instead require a custom Scheduler to implement ScheduledExecutorService? 🤔